relay-server/src/actors/project_cache.rs

```diff
@@ -185,126 +188,435 @@ impl Handler<GetProject> for ProjectCache {
+        let redis_source = self.redis_source.clone();
+
+        self.local_source
+            .send(FetchOptionalProjectState { project_key })
+            .map_err(|_| ())
+            .and_then(move |response| {
+                if let Some(state) = response {
+                    return Box::new(future::ok(ProjectStateResponse::new(state)))
+                        as ResponseFuture<_, _>;
+                }
+
+                match relay_mode {
+                    RelayMode::Proxy => {
+                        return Box::new(future::ok(ProjectStateResponse::new(Arc::new(
+                            ProjectState::allowed(),
+                        ))));
+                    }
+                    RelayMode::Static => {
+                        return Box::new(future::ok(ProjectStateResponse::new(Arc::new(
+                            ProjectState::missing(),
+                        ))));
+                    }
+                    RelayMode::Capture => {
+                        return Box::new(future::ok(ProjectStateResponse::new(Arc::new(
+                            ProjectState::allowed(),
+                        ))));
+                    }
+                    RelayMode::Managed => {
+                        // Proceed with loading the config from redis or upstream
+                    }
+                }
+
+                #[cfg(not(feature = "processing"))]
+                let fetch_redis = future::ok(None);
+
+                #[cfg(feature = "processing")]
+                let fetch_redis: ResponseFuture<_, _> = if let Some(ref redis_source) = redis_source
+                {
+                    Box::new(
+                        redis_source
+                            .send(FetchOptionalProjectState { project_key })
+                            .map_err(|_| ()),
+                    )
+                } else {
+                    Box::new(future::ok(None))
+                };
+
+                let fetch_redis = fetch_redis.and_then(move |response| {
+                    if let Some(state) = response {
+                        return Box::new(future::ok(ProjectStateResponse::new(state)))
+                            as ResponseFuture<_, _>;
+                    }
+
+                    let fetch_upstream = upstream_source
+                        .send(FetchProjectState {
+                            project_key,
+                            no_cache,
+                        })
+                        .map_err(|_| ())
+                        .and_then(move |result| result.map_err(|_| ()));
+
+                    Box::new(fetch_upstream)
+                });
+
+                Box::new(fetch_redis)
+            })
+            .into_actor(self)
+            .then(move |state_result, slf, context| {
+                let project = slf.get_or_create_project(project_key, context.address());
+                project.update_state(state_result.ok(), no_cache);
+                fut::ok::<_, (), _>(())
```

Comment:
Can you check if that works?

```suggestion
                fut::ok(())
```

Reply:
yep, worked... I think at some point I had to specify the return type